google/cloud/spanner_v1/session.py

```diff
@@ -113,7 +116,10 @@ def name(self):
         :raises ValueError: if session is not yet created
         """
         if self._session_id is None:
-            raise ValueError("No session ID set by back-end")
+            err_message = "No session ID set by back-end"
+            current_span = get_current_span()
+            current_span.add_event(err_message)
```

Comment:
Will this error not get captured by parent span? Do we need to add it as event here?

Reply:
The error shall be set but not the events on the span aka when it occured. However, now that you mention it, I was using it to debug and should perhaps just remove it entirely as it is non-standard of a span event/annotation.